testing/internal-integ-testing/src/test/groovy/org/gradle/test/fixtures/server/http/BlockingHttpServerTest.groovy

```diff
@@ -1290,7 +1291,13 @@ class BlockingHttpServerTest extends ConcurrentSpec {
         when:
         async {
             start {
-                succeeds("a")
+                // On Java 24, the URL connection will properly throw an IOException when trying to read the response (because of the fix for https://bugs.openjdk.org/browse/JDK-8335135)
+                // Expect it here, which will prevent noExceptionThrown() from triggering
+                if (JavaVersion.current() >= JavaVersion.VERSION_24) {
+                    fails("a")
```

Comment:
Maybe this should also assert something like:
```
assert fails("a").message == "Premature EOF"
```
So there aren't other kinds of failures that are also considered "OK"

Reply:
Done